servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/DefaultCompositeCloseable.java

```diff
@@ -203,8 +202,8 @@ private Completable buildCompletable(Function<AsyncCloseable, Completable> close
     }
 
     private static final class Operand {
-        private final List<AsyncCloseable> closables = new ArrayList<>(4);
-        private final boolean isMerge;
+        final Deque<AsyncCloseable> closables = new ArrayDeque<>(2);
```

Comment:
The default size of the collection changes. There is probably no reason not to still use 4 as that still typically fits within the alignment size of a heap allocation of a reference array. ie. The first resize to 4 will probably just reallocate to another array with less wasted space.